ruby/ext/google/protobuf_c/message.c

```diff
@@ -1263,13 +1263,17 @@ VALUE build_module_from_enumdesc(VALUE _enumdesc) {
   int n = upb_EnumDef_ValueCount(e);
   for (int i = 0; i < n; i++) {
     const upb_EnumValueDef* ev = upb_EnumDef_Value(e, i);
-    const char* name = upb_EnumValueDef_Name(ev);
+    char* name = strdup(upb_EnumValueDef_Name(ev));
     int32_t value = upb_EnumValueDef_Number(ev);
     if (name[0] < 'A' || name[0] > 'Z') {
-      rb_warn(
+      if (name[0] >= 'a' && name[0] <= 'z') {
+        name[0] -= 32; // auto capitalize
```

Comment:
This behavior change needs a unit test. To make the unit test pass across all Ruby interpreters, a corresponding change in the JRuby implementation will be required; see https://github.com/protocolbuffers/protobuf/blob/520c601c99012101c816b6ccc89e8d6fc28fdbb8/ruby/src/main/java/com/google/protobuf/jruby/RubyEnumDescriptor.java#L167-L177

Reply:
Add for JRuby at 3ac6b8716dee2db9b58d1c045e049598a5b9fb77.

Adapt tests at d30f0bb62a8d234c896770eb9d9c51d63fe1b90c.